Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -3074,17 +3072,28 @@ - (void)startRecordingVoiceMemo
 
     if (![self.audioRecorder prepareToRecord]) {
         DDLogError(@"%@ audioRecorder couldn't prepareToRecord.", self.tag);
-        [self resetRecordingVoiceMemo];
+        [self cancelVoiceMemo];
         OWSAssert(0);
         return;
     }
 
     if (![self.audioRecorder record]) {
         DDLogError(@"%@ audioRecorder couldn't record.", self.tag);
-        [self resetRecordingVoiceMemo];
+        [self cancelVoiceMemo];
         OWSAssert(0);
         return;
     }
+
+    if (session.recordPermission != AVAudioSessionRecordPermissionGranted) {
+        DDLogError(@"%@ we do not have recording permission.", self.tag);
+        [self cancelVoiceMemo];
+        [ViewControllerUtils
```

Comment:
I think we want something more actionable like this: https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/call/OutboundCallInitiator.swift#L52